MQuince/MQuince.Integration.Services.Implementation/MedicationsConsumptationService.cs

```diff
@@ -0,0 +1,110 @@
+﻿using iTextSharp.text;
+using iTextSharp.text.pdf;
+using MQuince.Integration.Entities;
+using MQuince.Integration.Repository.Contracts;
+using MQuince.Integration.Services.Constracts.DTO;
+using MQuince.Integration.Services.Constracts.IdentifiableDTO;
+using MQuince.Integration.Services.Constracts.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace MQuince.Integration.Services.Implementation
+{
+    public class MedicationsConsumptationService : IMedicationsConsumptionService
+    {
+        private readonly IMedicationsConsumptionRepository _medicationsConsumptionRepository;
+        IEnumerable<IdentifiableDTO<MedicationsConsumptionDTO>> listMedicationsByDate= new List<IdentifiableDTO<MedicationsConsumptionDTO>>();
+
+        public MedicationsConsumptationService(IMedicationsConsumptionRepository medicationsConsumptionReposotiry)
+        {
+            _medicationsConsumptionRepository = medicationsConsumptionReposotiry;
+        }
+
+        public Guid Create(MedicationsConsumptionDTO entityDTO)
+        {
+            MedicationsConsumption medicationsConsumption = CreateMedicationsConsumptionFromDTO(entityDTO);
+
+            _medicationsConsumptionRepository.Create(medicationsConsumption);
+
+            return medicationsConsumption.getKeyConsumtion;
+        }
+
+        public bool Delete(Guid id) => _medicationsConsumptionRepository.Delete(id);
+
+        public IEnumerable<IdentifiableDTO<MedicationsConsumptionDTO>> GetAll()
+            => _medicationsConsumptionRepository.GetAll().Select(c => CreateDTOFromMedicationsConsumption(c));
+
+        public IdentifiableDTO<MedicationsConsumptionDTO> GetByApi(Guid id) => CreateDTOFromMedicationsConsumption(_medicationsConsumptionRepository.GetById(id));
+
+
+        public IEnumerable<IdentifiableDTO<MedicationsConsumptionDTO>> GetConsumptionBetweenDates(DateTime from, DateTime to)
+        {
+            
+            foreach (IdentifiableDTO<MedicationsConsumptionDTO> medication in GetAll().ToList())
+            {
+                if (medication.EntityDTO.DateOfConsumtion >= from && medication.EntityDTO.DateOfConsumtion <= to)
+                {
+                    listMedicationsByDate.Append(medication);
+                }
+            }
+            
+            return listMedicationsByDate;
+
+
+        }
+        
+        public void GeneratePdf(DateDTO dto)
```

Comment:
This method could be refactored. Creating a new Class (eg. PdfGenerator) with couple of methods should do the job. But this is a lot better then in last commit.